src/main/java/org/pac4j/jax/rs/filters/ApplicationLogoutFilter.java

```diff
@@ -22,23 +20,32 @@
 @Priority(Priorities.AUTHORIZATION)
 public class ApplicationLogoutFilter extends AbstractFilter {
 
-    private ApplicationLogoutLogic<Object, JaxRsContext> applicationLogoutLogic = new DefaultApplicationLogoutLogic<>();
+    private static final JaxRsApplicationLogoutLogic<JaxRsContext> DEFAULT_LOGIC = new JaxRsApplicationLogoutLogic<>();
+
+    private ApplicationLogoutLogic<Object, JaxRsContext> applicationLogoutLogic;
 
     private String defaultUrl;
 
     private String logoutUrlPattern;
 
     public ApplicationLogoutFilter(Providers providers, Config config) {
         super(providers, config);
-        ((DefaultApplicationLogoutLogic<Object, JaxRsContext>) applicationLogoutLogic)
-                .setProfileManagerFactory(c -> new JaxRsProfileManager(c));
     }
 
     @Override
     protected void filter(JaxRsContext context) throws IOException {
-        CommonHelper.assertNotNull("applicationLogoutLogic", applicationLogoutLogic);
 
-        applicationLogoutLogic.perform(context, config, adapter(), context.getAbsolutePath(defaultUrl, false),
+        ApplicationLogoutLogic<Object, JaxRsContext> apl;
+
+        if (applicationLogoutLogic != null) {
```

Comment:
What's the use of setting an `applicationLogoutLogic` this way? Is the `config.getApplicationLogoutLogic()` not enough?
Because without it, it would be easier to do that in the constructor as you have the `config` component. This is what I did in Play: https://github.com/pac4j/play-pac4j/blob/master/src/main/java/org/pac4j/play/java/SecureAction.java#L70